internal/shard/shard_test.go

```
@@ -0,0 +1,272 @@
+	if shard.String() != expected {
+		t.Errorf("expected %v, got %v", expected, shard.String())
+	}
+}
+func TestFetchShardRepository(t *testing.T) {
+	repo := FetchShardRepository()
+
+	if repo == nil {
+		t.Errorf("expected non-nil repository, got nil")
+	}
+
+	if len(repo.shards) != 1 {
```

golangci-lint (1.62.2)

SA5011: possible nil pointer dereference

(staticcheck)